dts/bindings/spi/infineon,xmc4xxx-spi.yaml

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2022 Schlumberger
+# SPDX-License-Identifier: Apache-2.0
+
+description: INFINEON XMC4XXX SPI controller
+
+compatible: "infineon,xmc4xxx-spi"
+
+include: [spi-controller.yaml, pinctrl-device.yaml]
+
+properties:
+    reg:
+      required: true
+
+    miso-src:
```

Comment:
Can this not be handled via pinctrl?

Reply:
For xmc4xxx, the input multiplexers are on the peripheral side, and the output muxes are on the GPIO.
In the xmc pinctrl design it is only taking care of the output side. It didn't feel right to have peripheral driver code in pinctrl.

Initially, I thought about adding miso-src (or other input mux info) to the pinctrl dts node, and instead use this information in the peripheral driver, but this was not accepted in the pinctrl review
https://github.com/zephyrproject-rtos/zephyr/pull/47581#discussion_r917667843